crapssim/player.py

```diff
@@ -43,98 +43,95 @@ class Player:
     def __init__(self, bankroll: typing.SupportsFloat,
                  bet_strategy: STRATEGY_TYPE = passline,
                  name: str = "Player",
-                 unit: typing.SupportsFloat = 5,
-                 table: 'Table' = None):
+                 unit: typing.SupportsFloat = 5):
         self.bankroll: float = bankroll
         self.bet_strategy: STRATEGY_TYPE = bet_strategy
         self.strat_info: dict[str, typing.Any] = {}
         self.name: str = name
         self.unit: typing.SupportsFloat = unit
-        self.table = None
-        if table is not None:
-            self.sit_at_table(table)
-
         self.bets_on_table: list[Bet] = []
-        self.total_bet_amount: float = 0.0
+
+    @property
+    def total_bet_amount(self) -> float:
+        return sum(x.bet_amount for x in self.bets_on_table)
 
     def sit_at_table(self, table: "Table"):
         table.add_player(self)
 
-    def bet(self, bet_object: Bet) -> None:
-        if not bet_object.allowed(self):
-            return
-
-        if self.bankroll >= bet_object.bet_amount:
-            self.bankroll -= bet_object.bet_amount
-
-            if (bet_object.name, bet_object.subname) in [(b.name, b.subname) for b in self.bets_on_table]:
-                existing_bet: Bet = self.get_bet(bet_object.name, bet_object.subname)
-                existing_bet.bet_amount += bet_object.bet_amount
-            else:
-                bet_object.player = self
-                bet_object.table = self.table
-                self.bets_on_table.append(bet_object)
-
-            self.total_bet_amount += bet_object.bet_amount
-
-    def remove(self, bet_object: Bet) -> None:
-        if bet_object in self.bets_on_table and bet_object.removable:
-            self.bankroll += bet_object.bet_amount
-            self.bets_on_table.remove(bet_object)
-            self.total_bet_amount -= bet_object.bet_amount
-
-    def has_bet(self, *bets_to_check: str) -> bool:
-        """ returns True if bets_to_check and self.bets_on_table has at least one thing in common """
-        bet_names = {b.name for b in self.bets_on_table}
-        return bool(bet_names.intersection(bets_to_check))
-
-    def get_bet(self, bet_name: str, bet_subname: str = "") -> Bet:
-        """returns first betting object matching bet_name and bet_subname.
-        If bet_subname="Any", returns first betting object matching bet_name"""
-        if bet_subname == "Any":
-            bet_name_list: list[str] = [b.name for b in self.bets_on_table]
-            ind: int = bet_name_list.index(bet_name)
+    def place_bet(self, bet: Bet, table: "Table") -> None:

```

Comment:
Do you think would be any confusion with `place_bet()` and actual Place bets?  I think I considered something like this initially when writing the function.   I know that bet becomes both a verb and a noun which could be confusing as well.  Thoughts?

Reply:
Yeah, I can see where that could cause some confusion. How do you feel about add_bet?

Reply:
Good call, changed with [5806a70](https://github.com/skent259/crapssim/pull/20/commits/5806a7097273ede219f38227f1067b3016cbea58)

Reply:
Fixed with https://github.com/skent259/crapssim/pull/20/commits/d3f030a59f3e0a9a5f99b24e3734da4a2ff69565